client/components/button/index.jsx

```diff
@@ -3,49 +3,43 @@
 /**
  * External dependencies
  */
-import React from 'react';
-import assign from 'lodash/assign';
+import { PropTypes, createElement, PureComponent } from 'react';
 import classNames from 'classnames';
-import noop from 'lodash/noop';
-
-export default React.createClass( {
-
-	displayName: 'Button',
-
-	propTypes: {
-		disabled: React.PropTypes.bool,
-		compact: React.PropTypes.bool,
-		primary: React.PropTypes.bool,
-		scary: React.PropTypes.bool,
-		type: React.PropTypes.string,
-		href: React.PropTypes.string,
-		onClick: React.PropTypes.func,
-		borderless: React.PropTypes.bool
-	},
-
-	getDefaultProps() {
-		return {
-			disabled: false,
-			type: 'button',
-			onClick: noop,
-			borderless: false
-		};
-	},
+import { omit } from 'lodash';
+
+export default class Button extends PureComponent {
+	static propTypes = {
```

Comment:
do we have some preference to use `static` into the class body vs use Component.propTypes = {}?